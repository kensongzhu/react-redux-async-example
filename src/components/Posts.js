import React from "react";


export default class Posts extends React.Component {
    render() {
        const {posts} = this.props;
        return (
            <ul>
                {posts.map((post, i) => (<li key={i}>{post.title}</li>))}
            </ul>
        );
    }
}