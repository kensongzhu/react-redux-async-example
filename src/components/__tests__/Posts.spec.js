import React from "react";
import {shallow} from "enzyme";
import Posts from "../Posts";

const setup = (setupProps = []) => {
    const defaultProps = [
        {title: "subreddit 1"},
        {title: "subreddit 2"},
        {title: "subreddit 3"}
    ];
    const props = [...defaultProps, ...setupProps];
    const wrapper = shallow(<Posts posts={props}/>);

    return {
        props,
        wrapper
    }
};

describe("Posts", () => {
    test("renders without crashing", () => {
        const {wrapper} = setup();
        expect(wrapper).toMatchSnapshot();
    });
});
