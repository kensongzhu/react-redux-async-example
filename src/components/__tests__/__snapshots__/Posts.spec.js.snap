// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts renders without crashing 1`] = `
<ul>
  <li
    key="0"
  >
    subreddit 1
  </li>
  <li
    key="1"
  >
    subreddit 2
  </li>
  <li
    key="2"
  >
    subreddit 3
  </li>
</ul>
`;
